bot/exts/moderation/silence.py

```diff
@@ -93,87 +93,229 @@ async def _async_init(self) -> None:
 
     @commands.command(aliases=("unhush",))
-    async def unsilence(self, ctx: Context) -> None:
+    async def unsilence(self, ctx: Context, *, channel: Union[TextChannel, VoiceChannel] = None) -> None:
         """
-        Unsilence the current channel.
+        Unsilence the given channel if given, else the current one.
 
         If the channel was silenced indefinitely, notifications for the channel will stop.
         """
         await self._init_task
-        log.debug(f"Unsilencing channel #{ctx.channel} from {ctx.author}'s command.")
-        await self._unsilence_wrapper(ctx.channel)
+        if channel is None:
+            channel = ctx.channel
+        log.debug(f"Unsilencing channel #{channel} from {ctx.author}'s command.")
+        await self._unsilence_wrapper(channel, ctx)
 
     @lock_arg(LOCK_NAMESPACE, "channel", raise_error=True)
-    async def _unsilence_wrapper(self, channel: TextChannel) -> None:
+    async def _unsilence_wrapper(
+        self, channel: Union[TextChannel, VoiceChannel], ctx: Optional[Context] = None
+    ) -> None:
         """Unsilence `channel` and send a success/failure message."""
+        msg_channel = channel
+        if ctx is not None:
+            msg_channel = ctx.channel
+
         if not await self._unsilence(channel):
-            overwrite = channel.overwrites_for(self._verified_role)
-            if overwrite.send_messages is False or overwrite.add_reactions is False:
-                await channel.send(MSG_UNSILENCE_MANUAL)
+            if isinstance(channel, VoiceChannel):
+                overwrite = channel.overwrites_for(self._verified_voice_role)
+                manual = overwrite.speak is False
             else:
-                await channel.send(MSG_UNSILENCE_FAIL)
+                overwrite = channel.overwrites_for(self._verified_msg_role)
+                manual = overwrite.send_messages is False or overwrite.add_reactions is False
+
+            # Send fail message to muted channel or voice chat channel, and invocation channel
+            if manual:
+                await self.send_message(MSG_UNSILENCE_MANUAL, msg_channel, channel)
+            else:
+                await self.send_message(MSG_UNSILENCE_FAIL, msg_channel, channel)
+
         else:
-            await channel.send(MSG_UNSILENCE_SUCCESS)
+            await self.send_message(MSG_UNSILENCE_SUCCESS, msg_channel, channel, True)
 
-    async def _set_silence_overwrites(self, channel: TextChannel) -> bool:
+    async def _set_silence_overwrites(self, channel: Union[TextChannel, VoiceChannel], kick: bool = False) -> bool:
         """Set silence permission overwrites for `channel` and return True if successful."""
-        overwrite = channel.overwrites_for(self._verified_role)
-        prev_overwrites = dict(send_messages=overwrite.send_messages, add_reactions=overwrite.add_reactions)
+        is_text_channel = isinstance(channel, TextChannel)
```

Comment:
I think you misunderstood what I was aiming for. The problem is that there are if statements that are avoidable. The problem is not with calling `isinstance` twice. What you've done still keeps the avoidable if statements. I've corrected my original code suggestion for this (https://github.com/python-discord/bot/pull/1297#discussion_r529262670) and I still think you should implement it like that.

Reply:
I definitely did, I'll go ahead and implement that correctly.